assets/src/components/component/ComponentInfo.tsx

```diff
@@ -31,31 +31,25 @@ function hasPods(kind: string): boolean {
   return componentsWithPods.includes(kind)
 }
 
+const componentInfoMap: { [key: string]: JSX.Element } = {
+  certificate: <Certificate />,
+  cronjob: <CronJob />,
+  deployment: <Deployment />,
+  ingress: <Ingress />,
+  job: <Job />,
+  service: <Service />,
+  statefulset: <StatefulSet />,
+  daemonset: <DaemonSet />,
+  canary: <CanaryInfo />,
+  servicedeployment: <PluralServiceDeployment />,
+}
+
 function getInfo(kind: string): JSX.Element | undefined {
-  switch (kind) {
-    case 'certificate':
-      return <Certificate />
-    case 'cronjob':
-      return <CronJob />
-    case 'deployment':
-      return <Deployment />
-    case 'ingress':
-      return <Ingress />
-    case 'job':
-      return <Job />
-    case 'service':
-      return <Service />
-    case 'statefulset':
-      return <StatefulSet />
-    case 'daemonset':
-      return <DaemonSet />
-    case 'canary':
-      return <CanaryInfo />
-    case 'servicedeployment':
-      return <PluralServiceDeployment />
-    default:
-      return undefined
-  }
+  return componentInfoMap[kind]
```

Comment:
Did you test this with the other component views?  I'd be somewhat concerned it behaves weirdly given that contexts play such a huge role in those components

Reply:
Yeah I did, didn't find any issues. Context shouldn't be affected because the component tree hasn't changed